linode/token/framework_model.go

```diff
@@ -14,21 +14,17 @@ import (
 type ResourceModel struct {
 	Label   types.String                        `tfsdk:"label"`
 	Scopes  customtypes.LinodeScopesStringValue `tfsdk:"scopes"`
-	Expiry  customtypes.RFC3339TimeStringValue  `tfsdk:"expiry"`
-	Created customtypes.RFC3339TimeStringValue  `tfsdk:"created"`
+	Expiry  timetypes.RFC3339                   `tfsdk:"expiry"`
+	Created timetypes.RFC3339                   `tfsdk:"created"`
```

Comment:
Tangentially related to this change, it looks like the creation logic now fails if an expiry isn't explicitly defined: https://github.com/linode/terraform-provider-linode/blob/c5578aca7f01109609d4c60760f3a1ca42d73a0f/linode/token/framework_resource.go#L43C8-L55

This can be resolved with something like this:

```go
	var expiry *time.Time

	if !data.Expiry.IsNull() {
		parsedExpiry, d := data.Expiry.ValueRFC3339Time()
		resp.Diagnostics.Append(d...)
		if d.HasError() {
			return
		}

		expiry = &parsedExpiry
	}

	createOpts := linodego.TokenCreateOptions{
		Label:  data.Label.ValueString(),
		Scopes: data.Scopes.ValueString(),
		Expiry: expiry,
	}
```

Reply:
Thanks for catching that issue!